### ISSUE TITLE
I wish I could preview and audit my release size

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#13189]
Similar-ish to https://github.com/flutter/flutter/issues/6177.

Today I can use `flutter build aot` and `du -sch build/aot`:

> 6.4M	build/aot
> 6.4M	total

It would be nice to have a way to audit the size of my binary, for example `flutter audit`:

> 6.4M
>
> ICU Data:       x.xM
> Engine:          x.xM
> Framework:   x.x.M
> Packages:     x.xM
> 
> package:flutter_foo_plugin x.xKb
> ...

Otherwise I could easily see my binary hitting 10s of Mbs, with no practical way to reduce without commenting out code/assets, rebuilding, and trying again.

## Steps to Reproduce

Basically just the default application of `flutter create` then:
* `flutter build aot`
* `du -sch build/aot`

**Comment by sethladd** [comment]
Thanks for the feedback!

Consider this another vote for an extensible task/build system. `flutter_tool` can't be all things to all people, instead it should be extensible by our ecosystem. I don't have a solution in mind, but I do know that a logical next step for Flutter's ecosystem is an extensible task runner of some sort. Something that would allow users to install new tasks (like `audit`) from pub and then run `flutter audit`.

**Comment by sethladd** [comment]
cc @mit-mit @devoncarew

**Comment by mit-mit** [comment]
Yes, some kind of extensibility model has been brought on many times. I opened https://github.com/flutter/flutter/issues/13834 to track it.

**Comment by mit-mit** [comment]
Punting the present issue to milestone 4; this doesn't block beta.

**Comment by eseidelGoogle** [comment]
https://github.com/flutter/flutter/issues/16191 might be an alternative?

**Comment by jonahwilliams** [comment]
This can now be done (as much as is possible with iOS) via the new devtools code size tooling: https://flutter.dev/docs/development/tools/devtools/app-size
